transcendence-app/src/user/user.controller.ts

```diff
@@ -33,6 +36,21 @@ export class UserController {
     return req.user;
   }
 
+  @Get('users')
+  getUsers() {
+    return this.userService.retrieveUsers();
+  }
+
+  @Get('usernames?')
+  getUsernames(
+    @Query('take', ParseIntPipe) take: number,
+    @Query('skip', ParseIntPipe) skip: number,
+    @Query('sort', ParseBoolPipe) sort: boolean,
+  ) {
+    take = take > 20 ? 20 : take;
```

Comment:
This logic is a little bit weird and I'd say no to have magic numbers -> https://en.wikipedia.org/wiki/Magic_number_(programming)

Reply:
Normally you name them appropriately (which is hard), assign them to some constant/define/enum, and use the named constant/define/enum instead.
To me, here, it seems like a "MAX_USER_ENTRIES_PER_PAGE" or "MaxUserEntriesPerPage" (depending on our naming convention which I'm not sure about right now) kind of thing (I'm guessing)?
If you are doing a max, don't hand roll it with a ternary, use Math.max() and such.

Reply:
You are right. I correct it.